container/container_unit_test.go

```diff
@@ -66,3 +70,54 @@ func TestContainerSecretReferenceDestTarget(t *testing.T) {
 		t.Fatalf("expected secret dest %q; received %q", expected, d)
 	}
 }
+
+func TestContainerLogPathSetForJSONFileLogger(t *testing.T) {
+	rootPath := filepath.Join(string(filepath.Separator), "root")
+	os.MkdirAll(filepath.Join(rootPath, "root"), os.ModePerm)
+	defer os.RemoveAll(rootPath)
+
+	c := &Container{
+		Config: &container.Config{},
+		HostConfig: &container.HostConfig{
+			LogConfig: container.LogConfig{
+				Type: jsonfilelog.Name,
+			},
+		},
+		ID:   filepath.Join("root", "json-file-logger"),
+		Root: rootPath,
+	}
+
+	_, err := c.StartLogger()
+	require.NoError(t, err)
+
+	expectedLogPath, err := filepath.Abs(filepath.Join(rootPath, fmt.Sprintf("%s-json.log", c.ID)))
+	require.NoError(t, err)
+	require.Equal(t, c.LogPath, expectedLogPath)
+}
+
+func TestContainerLogPathSetForRingLogger(t *testing.T) {
+	rootPath := filepath.Join(string(filepath.Separator), "root")
+	os.MkdirAll(filepath.Join(rootPath, "root"), os.ModePerm)
```

Comment:
Why not `ioutil.TempDir()`? This looks like it's creating a test dir in `/root/root`?

Reply:
It looks like the build passed using a `TempDir`, let me know if there's any other changes you'd like here